plonky2x/core/src/frontend/eth/rlp/utils.rs

```diff
@@ -0,0 +1,379 @@
+use ethers::types::Bytes;
+use log::info;
+use num::bigint::ToBigInt;
+use num::BigInt;
+
+/// Byte array of at most 32 bytes, potentially padded. This could be a hash, value,
+/// rlp.encode(branch node), rlp.encode(leaf node), or rlp.encode(extension node) under the hood.
+pub const MAX_STRING_SIZE: usize = 32;
+pub type FixedSizeString = [u8; MAX_STRING_SIZE];
```

Comment:
Perhaps FixedSizeString should actually be the following and encode both the data and the length.

struct FixedSizeString<const L: usize> {
    data: [u8; L];
    len: usize
};

Reply:
And then maybe you should make an impl on the struct such that you can do

let s: FixedSizeString;
let metadata: RLPEncodingMetadata = s.rlp_encoding_metadata();

Reply:
In general, we should try to make things more object oriented